src/app/core/services/issue.service.ts

```diff
@@ -44,12 +46,20 @@ export class IssueService {
    *
    */
   getAllIssues(): Observable<Issue[]> {
-    if (this.issues === undefined) {
+    if (this.issues.value === undefined) {
       return this.initializeData();
     }
     return this.issues$;
   }
 
+  reloadAllIssues() {
+    return this.initializeData().subscribe();
```

Comment:
This function is used by the header component. So ideally, this function should return an observable. By doing so, in the header component you can do something like this.

```typescript
this.issueService.reloadAllIssues.subscribe(
    (success) => {},
    (error) => this.errorHandlingService.handleHttpError(error);
)
```

The main point is that we rarely just `.subscribe()` and leave it hanging there. We usually coupled it with a success callback function and error handling function. And those functions usually come with changes in the UI (Because we want to show some feedback to user) so we will usually declare `.subscribe()` to the components, since the Components are the one responsible for UI, **not** Service.

Reply:
Noted, I have added the error handling

Reply:
Can subscribe to the reload function from the header component? I have explained above. Services should be used for manipulating/obtaining data not UI.

Reply:
Can subscribe to the reload function from the header component? I have explained above. Services should be used for manipulating/obtaining data not UI.

Reply:
Hi, sorry about that, I have moved it to the correct component